### PR TITLE
Document Hailo INT8 accuracy expectations and deployment guidance

### PR DESCRIPTION
**Pull request by JESUSROYETH**
## Summary

Follow-up to the quantization findings from #25247, where @glenn-jocher suggested moving them into the docs. This adds a new "Accuracy Expectations by Model Family" subsection to the Hailo integration page, with the measured INT8 mAP50 retention per family and the three practical rules that come from the device measurements:

1. Calibration has to be in-domain — out-of-domain fine-tuning gave the same result as no fine-tuning at all.
2. YOLO26 confidence shifts on device, `conf` ends about 0.05 lower than in PyTorch.
3. Attention layers lose more accuracy on Hailo-8/8L, because the DFC only supports 8-bit matmul inputs on this generation.

The subsection also states the evaluation protocol explicitly: both models compared at the same `conf`. YOLOv8/YOLO11 HEFs run the on-chip NMS with the threshold fixed at export time, so comparing against a PyTorch `val` run at its default low threshold makes the quantization gap look bigger than it is. Without that note the table numbers are hard to reproduce.

Also, the "Quantized Model Accuracy Drops" troubleshooting entry now links to the table, so a user can check if their gap is normal for that family or a real calibration problem.

All numbers come from the Hailo-8L measurements reported in the #25247 discussion (HailoRT 4.23, DFC 3.33, COCO128 protocol).

Deleted: nothing — one docs subsection plus one troubleshooting cross-reference added, no code changes.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Documents Hailo INT8 accuracy expectations and deployment guidance across YOLO model families.

### 📊 Key Changes
- Adds measured mAP50 retention baselines for YOLOv8n, YOLO11n, and YOLO26n on Hailo-8L with in-domain calibration.
- Explains confidence-threshold alignment between PyTorch evaluation and on-chip NMS for exported HEFs.
- Provides practical guidance on in-domain calibration, YOLO26 confidence adjustment, and attention-related quantization limits on Hailo-8/8L.
- Updates the quantized accuracy troubleshooting section to reference the new model-family baselines.

### 🎯 Purpose & Impact
- Sets clearer expectations for INT8 accuracy before deployment.
- Helps users choose calibration data and confidence thresholds that better preserve device performance.
- Highlights model-family and hardware considerations that can affect Hailo deployment accuracy.
- Improves documentation for developers deploying Ultralytics models to Hailo edge devices.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JESUSROYETH, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms receipt, and an Ultralytics engineer will assist with the review. Please review the following checklist:
- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: docs/en/integrations/hailo.md [hunk 1]
```diff
@@ -165,6 +165,24 @@ model.export(format="hailo", name="hailo8l", data="my_dataset.yaml")
...
+
+Three practical rules follow from device measurements:
+
+1. **Calibrate in-domain, always.** Fine-tuning with out-of-domain images is equivalent to disabling fine-tuning entirely: a YOLO26n calibrated with 1,238 out-of-domain images retains the same accuracy (85.7%) as one compiled without fine-tuning. A small in-domain set beats a large out-of-domain one.
+2. **Lower `conf` by about 0.05 for YOLO26 deployments.** Quantization shifts YOLO26 scores down by roughly 0.05 on average, so a threshold tuned in PyTorch drops valid detections on the HEF. Using `conf=0.20` on device matches the detection count of PyTorch at `conf=0.25` and recovers about half of the accuracy gap; the remainder is ranking noise that no threshold recovers.
+3. **The attention penalty is structural on Hailo-8/8L.** The Dataflow Compiler only supports 8-bit inputs for `matmul` layers on this hardware generation, so attention products cannot run at higher precision. When accuracy is the top priority and the model choice is flexible, YOLO11 currently quantizes better than YOLO26 on Hailo-8/8L.
```
Comment (UltralyticsAssistant) [inline docs/en/integrations/hailo.md]: 💡 **MEDIUM**: This attributes the precision restriction to `matmul` inputs, but Hailo supports configuring matmul layers for `a16_w16`; the documented forced 8-bit transition in attention paths is associated with the elementwise multiplication operation. As written, this gives users the wrong hardware limitation and may prevent them from trying supported mixed-precision configurations. Please correct the operator-specific explanation and scope it to the tested DFC/hardware version.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The new calibration and threshold guidance is useful, but the attention-precision explanation needs correction before publishing.

💬 Posted 1 inline comment

File: docs/en/integrations/hailo.md (hunk 1 above)
Comment (JESUSROYETH) [reply docs/en/integrations/hailo.md]: Thanks, I tightened the wording and scoped it to DFC 3.33.

On the operator detail, in case it is useful: on the DFC 3.33 I used for the tests, the matmul layer only accepts `a8_w8`, `a8_w8_a8` and `a8_w8_a16` (`hailo_matmul.py`), so the activation inputs are INT8 in every mode, there is no `a16_w16` for it. The 16-bit-output mode does exist but it fails allocation for this graph. And in YOLO26 the attention path translates to matmul, not elementwise multiplication — the body graph has `matmul1..4` and no `ew_mult` — so the INT8 step is on the matmul inputs. Protecting the surrounding convolutions at 16-bit did not change the mAP, which fits that.
